control-plane/agents/src/bin/core/controller/scheduling/volume_policy/pool.rs

```diff
@@ -73,40 +73,88 @@ impl PoolBaseFilters {
                     PoolTopology::Labelled(labelled_topology) => {
-                        // The labels in Volume Pool Topology should match the pool labels if
-                        // present, otherwise selection of any pool is allowed.
-                        if !labelled_topology.inclusion.is_empty() {
-                            volume_pool_topology_labels = labelled_topology.inclusion
+                        // Return false if the exclusion and incluson labels has any common key.
+                        if labelled_topology
+                            .inclusion
+                            .keys()
+                            .any(|key| labelled_topology.exclusion.contains_key(key))
+                        {
+                            return false;
+                        }
+
+                        if !labelled_topology.inclusion.is_empty()
+                            || !labelled_topology.exclusion.is_empty()
+                        {
+                            volume_pool_topology_inclusion_labels = labelled_topology.inclusion;
+                            volume_pool_topology_exclusion_labels = labelled_topology.exclusion;
                         } else {
                             return true;
                         }
                     }
                 },
             },
         };
-        // We will reach this part of code only if the volume has pool topology labels.
+
+        // We will reach this part of code only if the volume has inclusion/exclusion labels.
         match request.registry().specs().pool(&item.pool.id) {
-            Ok(spec) => match spec.labels {
-                None => false,
-                Some(label) => volume_pool_topology_labels
-                    .iter()
-                    .all(|(vol_key, vol_val)| {
-                        // See `InclusiveLabel` doc comment.
-                        // todo: add exclusion
-                        label
-                            .get(vol_key)
-                            .is_some_and(|pool_value| vol_val.is_empty() || pool_value == vol_val)
-                    }),
-            },
+            Ok(spec) => {
+                match spec.labels {
+                    None => false,
+                    Some(label) => {
+                        // The inclusion labes of volume must match the labels in pool
+                        // for the specified pool to qualify for volume provisioning.
+                        // i.e both the maps volume_pool_topology_inclusion_labels and label must be
+                        // equal
+                        let inclusion_match = label.iter().all(|(key, value)| {
+                            volume_pool_topology_inclusion_labels.get(key) == Some(value)
+                        }) && volume_pool_topology_inclusion_labels
+                            .iter()
+                            .all(|(key, value)| label.get(key) == Some(value));
```

Comment:
Why is this required to match?

```suggestion
                        });
```